### PR TITLE
api.main: fix user authentication

### PR DESCRIPTION
**Pull request by JenySadadia**
Use `main.get_password_hash` instead of
`auth.get_password_hash` directly to convert
password from SecretStr to plain text first.

Fixes: 7a67489 ("api/main.py: implement /user endpoint")
Signed-off-by: Jeny Sadadia <jeny.sadadia@collabora.com>

### CODE REVIEW

**Comment by mgalka** [comment]
@JenySadadia Do you have this change tested?

**Comment by JenySadadia** [comment]
> @JenySadadia Do you have this change tested?

Yes, tested OK on local.

File: api/main.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ async def post_user(
         current_user: User = Security(get_user, scopes=["admin"])):
     """Create new user"""
     try:
-        hashed_password = auth.get_password_hash(str(password.password))
+        hashed_password = get_password_hash(password)
```
Comment (gctucker) [inline api/main.py:78]: The issue is that `get_password_hash()` is an API entry point function, so I don't think it's a good idea to call it from within another entry point function.

This should work too and would stay within the current layer:
```python
hashed_password = auth.get_password_hash(password.password.get_secret_value())
```

Comment (JenySadadia) [reply api/main.py]: Done.

**Pull request merged**